### PR TITLE
Minto Stuff - Bringing up Asset Editor back to stable part 1

### PR DESCRIPTION
**Pull request by mr-phazer**
Do you get it?

### CODE REVIEW

**Comment by admiralnelson** [comment]
Ja!

File: AssetEditor/AssetEditor.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
         <TargetFramework>net6.0-windows</TargetFramework>
         <UseWPF>true</UseWPF>
         <ApplicationIcon>hnet.com-image (1).ico</ApplicationIcon>
-        <AssemblyName>AssetEditor v0.45</AssemblyName>
+        <AssemblyName>AssetEditor45-PhazerEdition</AssemblyName>
```
Comment (admiralnelson) [inline AssetEditor/AssetEditor.csproj:7]: It's better idea to call it AssetEditor v.05 instead to avoid any confusion with old userbase. Also for historical reason.

File: AssetEditor/AssetEditor.csproj [hunk 2]
```diff
@@ -53,13 +55,13 @@
...
+        <Product>AssetEditor45-Phazt{Tester of FBX Model impor/export}</Product>
         <PackageProjectUrl>https://github.com/donkeyProgramming/TheAssetEditor</PackageProjectUrl>
         <RepositoryUrl>https://github.com/donkeyProgramming/TheAssetEditor</RepositoryUrl>
         <RepositoryType>GIT</RepositoryType>
-        <AssemblyVersion>0.45.0.0</AssemblyVersion>
-        <FileVersion>0.45.0.0</FileVersion>
-        <PackageId>AssetEditor v0.45</PackageId>
+        <AssemblyVersion>0.2.0.0</AssemblyVersion>
+        <FileVersion>0.2.0.0</FileVersion>
+        <PackageId>AssetEditor45-Phazt</PackageId>
```
Comment (admiralnelson) [inline AssetEditor/AssetEditor.csproj:64]: It's better idea to call it AssetEditor v.05 instead to avoid any confusion with old userbase. Also for historical reason.

File: AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-    <PublishReadyToRun>false</PublishReadyToRun>
-  </PropertyGroup>
+﻿<Project ToolsVersion="4.0">
+    <PropertyGroup>
+        <Configuration>Release</Configuration>
+        <Platform>Any CPU</Platform>
+        <PublishDir>bin\Release\publish\</PublishDir>
+        <PublishProtocol>FileSystem</PublishProtocol>
+        <TargetFramework>net6.0-windows</TargetFramework>
+        <SelfContained>false</SelfContained>
```
Comment (admiralnelson) [inline AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml:8]: should be set to true, so that .net dlls are baked in the program itself. 
for c++ DLL, we can include it as resource string later on and when the user run the program, the program will unpack it from its own internal resouce and loads them as if they were next to program itself (this part is low prio)

File: AssetEditor/Views/MainWindow.xaml.cs [hunk 4]
```diff
@@ -14,14 +16,19 @@ public partial class MainWindow : Window
...
-            System.Reflection.Assembly assembly = System.Reflection.Assembly.GetExecutingAssembly();
-            var fvi = System.Diagnostics.FileVersionInfo.GetVersionInfo(assembly.Location);
+            
+
+            //System.Reflection.Assembly assembly = System.Reflection.Assembly.GetExecutingAssembly();
+            FileVersionInfo fvi = FileVersionInfo.GetVersionInfo(System.Diagnostics.Process.GetCurrentProcess().MainModule.FileName);
+
+            Title = $"{fvi.ProductName} - {fvi.FileVersion}";
 
-            Title = $"AssetEditor v{fvi.FileMajorPart}.{fvi.FileMinorPart}";
```
Comment (admiralnelson) [inline AssetEditor/Views/MainWindow.xaml.cs:24]: Revert to the old commit to avoid any confusion with older userbase. also for historical preservation too

**Review by admiralnelson — CHANGES_REQUESTED** [review]
it's quite a big commit, I will take a look later on

File: AssetEditor/AssetEditor.csproj (hunk 1 above)
Comment (admiralnelson) [reply AssetEditor/AssetEditor.csproj:7]: also we can add our own credit in about window

Comment (mr-phazer) [reply AssetEditor/AssetEditor.csproj:7]: AssetEditor-Amiral-Phazer editor

File: AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml (hunk 3 above)
Comment (mr-phazer) [reply AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml:8]: Yes, I tried that, but I can try again, and did not work, or something, I can try again¨'

File: AssetEditor/Views/MainWindow.xaml.cs (hunk 4 above)
Comment (mr-phazer) [reply AssetEditor/Views/MainWindow.xaml.cs:24]: Ok, revert it all the back ole's origina?
Can we do "build numbers":)?

Comment (admiralnelson) [reply AssetEditor/Views/MainWindow.xaml.cs:24]: well, git doesn't have "revision number" like SVN does. but we can include build commit. here's how I achieved that https://github.com/admiralnelson/bannerlord-error-detector/blob/50f213a6f9418697fb638132d5a0e170727fbc06/Util.vb#L385   
Basically there's inert file called commit.txt which contains the latest commit hash, this file will be generated automatically by visual studio everytime we hit compile button.

File: AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml (hunk 3 above)
Comment (admiralnelson) [reply AssetEditor/Properties/PublishProfiles/FolderProfile.pubxml:8]: ok then, we can leave it as false for now
We should come back again, and set this back to true. (high prio)
I will make issue about this to remind ourselves.

File: AssetManagment/GeometryManagement/MeshImportExporter.cs
Comment (admiralnelson) [inline AssetManagment/GeometryManagement/MeshImportExporter.cs:1]: please add notice or comment on this file on why it was disabled.

**Review by admiralnelson — CHANGES_REQUESTED** [review]

**Comment by admiralnelson** [comment]
I have no comment on the C++ side of things, they look fine to me.

File: .github/workflows/pr-build.yml
Comment (admiralnelson) [inline .github/workflows/pr-build.yml:1]: Related issue nr https://github.com/admiralnelson/TheAssetEditor/issues/3

File: AssetEditor/AssetEditor.csproj (hunk 1 above)
Comment (admiralnelson) [reply AssetEditor/AssetEditor.csproj:7]: Just leave it AssetEditor v0.5, we can build our about dialog with our names and credit on it.
[Resolve this and then I can merge it]

File: AssetEditor/Services/VersionChecker.cs [hunk 5]
```diff
@@ -18,8 +18,10 @@ public static void CheckVersion()
 
             try
             {
-                System.Reflection.Assembly assembly = System.Reflection.Assembly.GetExecutingAssembly();
-                FileVersionInfo fvi = FileVersionInfo.GetVersionInfo(assembly.Location);
+                //System.Reflection.Assembly assembly = System.Reflection.Assembly.GetExecutingAssembly();
```
Comment (admiralnelson) [inline AssetEditor/Services/VersionChecker.cs:21]: why was this changed? What's wrong with the old way of retrieving file version info on the .exe?

**Review by admiralnelson — CHANGES_REQUESTED** [review]

File: AssetEditor/AssetEditor.csproj (hunk 2 above)
Comment (admiralnelson) [reply AssetEditor/AssetEditor.csproj:64]: Related issue (low prio) https://github.com/admiralnelson/TheAssetEditor/issues/5

Comment (admiralnelson) [reply AssetEditor/AssetEditor.csproj:64]: Update from discord convo: it should be called AssetEditor45 (45 because we started on version 45) AssetEditor (because we want to keep it generic and open)
[Resolve this and then I can merge it]

File: AssetEditor/Views/MainWindow.xaml.cs (hunk 4 above)
Comment (admiralnelson) [reply AssetEditor/Views/MainWindow.xaml.cs:24]: Update from discord convo: it should be called AssEd45 (45 because we started on version 45) AssEd (because we want to keep it generic and open, also it sounds kinda funny)
[Resolve this and then I can merge it]

File: AssetEditor/AssetEditor.csproj (hunk 1 above)
Comment (mr-phazer) [reply AssetEditor/AssetEditor.csproj:7]: AssEd45 (version 0.2.1.1)

File: AssetEditor/Views/MainWindow.xaml.cs (hunk 4 above)
Comment (mr-phazer) [reply AssetEditor/Views/MainWindow.xaml.cs:24]: Done.

**Comment by admiralnelson** [comment]
Done!
